### PR TITLE
Refactor analysis smoother update

### PR DESCRIPTION
**Pull request by ManInFez**
This is the first step of https://github.com/equinor/ert/issues/2887

It also contains some cleanup of unused options of the analysis module.

## Pre review checklist

- [x] Added appropriate labels

Adding labels helps the maintainers when writing release notes, see sections and the
corresponding labels here: https://github.com/equinor/ert/blob/main/.github/release.yml

### CODE REVIEW

File: libres/lib/analysis/analysis_module.cpp [hunk 1]
```diff
@@ -100,26 +101,23 @@ static UTIL_SAFE_CAST_FUNCTION(analysis_module, ANALYSIS_MODULE_TYPE_ID)
 
 void analysis_module_initX(analysis_module_type *module, matrix_type *X,
```
Comment (ManInFez) [inline libres/lib/analysis/analysis_module.cpp:102]: We could also remove these functions as they are only used a test

**Comment by ManInFez** [comment]
test ctest please!
test ert please!

**Comment by ManInFez** [comment]
test ert please!

File: libres/lib/analysis/analysis_module.cpp (hunk 1 above)
Comment (joakim-hove) [reply libres/lib/analysis/analysis_module.cpp:102]: Yes!

Comment (joakim-hove) [reply libres/lib/analysis/analysis_module.cpp]: Done

**Comment by joakim-hove** [comment]
I have replayed the rebase here: https://github.com/equinor/ert/pull/2893

**Comment by ManInFez** [comment]
fixed version fpound here: https://github.com/equinor/ert/pull/2893

**Pull request closed without merging**